### PR TITLE
Add tags automatically for difficulty if it is not set to undefined

### PR DESCRIPTION
**Pull request by Bluesy1**

### CODE REVIEW

File: tests/test_question_templates/question_expected_outputs/instructor/q06_number-input_feedback/q06_number-input_feedback.md [hunk 1]
```diff
@@ -117,7 +117,7 @@ myst:
       v: '2.50'
       p: '4'
     correct_answers:
-      part1_ans: !!python/object/apply:numpy.core.multiarray.scalar
+      part1_ans: !!python/object/apply:numpy._core.multiarray.scalar
```
Comment (Bluesy1) [inline tests/test_question_templates/question_expected_outputs/instructor/q06_number-input_feedback/q06_number-input_feedback.md:120]:
```suggestion
      part1_ans: !!python/object/apply:numpy.core.multiarray.scalar
```

File: src/problem_bank_scripts/problem_bank_scripts.py [hunk 2]
```diff
@@ -475,6 +467,11 @@ def write_info_json(output_path, parsed_question):
         auto_tags.append("multi_part")
     auto_tags.extend(list(set(q_types)))
 
+    if (difficulties := parsed_question["header"].get("difficulty", ["undefined"])) != ["undefined"]:
+        if not isinstance(difficulties, list):
+            difficulties = [difficulties]
+        auto_tags.extend(f"difficulty-{difficulty.lower()}" for difficulty in difficulties)
```
Comment (firasm) [inline src/problem_bank_scripts/problem_bank_scripts.py:473]:
```suggestion
        auto_tags.extend(f"{difficulty.lower()}" for difficulty in difficulties)
```

Thanks for submitting this enhancement ! I'm in favour but I think the difficulty- is redundant and is unnecessary/redundant. It'll also not look great in PL with the tags front end.

File: tests/test_question_templates/question_expected_outputs/prairielearn-dev/q18_integer-input/info.json [hunk 3]
```diff
@@ -1,9 +1,10 @@
 {
-    "uuid": "03d2fc31-cc01-3a77-9394-05da8968db50",
+    "uuid": "fd488716-df71-3a46-8d65-e64a08d1ee07",
     "title": "Median",
     "topic": "000.Template",
     "tags": [
         "integer-input",
+        "difficulty-easy",
```
Comment (Bluesy1) [inline tests/test_question_templates/question_expected_outputs/prairielearn-dev/q18_integer-input/info.json:7]:
```suggestion
        "easy",
```

File: tests/test_question_templates/question_expected_outputs/prairielearn/q18_integer-input/info.json [hunk 4]
```diff
@@ -1,9 +1,10 @@
 {
-    "uuid": "9880f3fb-8d29-3365-908c-d2a4eec045c7",
+    "uuid": "999970d4-630b-38de-a43b-dbba13af980e",
     "title": "Median",
     "topic": "000.Template",
     "tags": [
-        "integer-input"
+        "integer-input",
+        "difficulty-easy"
```
Comment (Bluesy1) [inline tests/test_question_templates/question_expected_outputs/prairielearn/q18_integer-input/info.json:7]:
```suggestion
        "easy"
```

File: src/problem_bank_scripts/problem_bank_scripts.py (hunk 2 above)
Comment (Bluesy1) [reply src/problem_bank_scripts/problem_bank_scripts.py:473]: Fair enough! If we're removing the prefix, we don't need the f-string at all:
```suggestion
        auto_tags.extend(difficulty.lower() for difficulty in difficulties)
```

File: src/problem_bank_scripts/problem_bank_scripts.py [hunk 5]
```diff
@@ -329,29 +329,21 @@ def read_md_problem(filepath):
...
-                    x,
-                ]
+                blocks[f"block{block_count}"] = [x]
             else:
-                blocks["block{0}".format(block_count - 1)].append(x)
-                blocks["block{0}".format(block_count)] = [
-                    x,
-                ]
+                blocks[f"block{block_count - 1}"].append(x)
+                blocks[f"block{block_count}"] = [x]
```
Comment (Bluesy1) [inline src/problem_bank_scripts/problem_bank_scripts.py:335]: I missed that my autoformatter/linter in vscode for other projects changed some things here ☹️  - I can revert if needed

**Pull request merged**